pages/_document.js

```diff
@@ -51,7 +56,23 @@ class MyDocument extends Document {
             content={pageContext ? pageContext.theme.palette.primary.main : null}
           />
           <link rel="shortcut icon" href="/static/favicon.ico" />
-          <link rel="canonical" href={canonical} />
+          {/* SEO */}
+          <link
+            rel="canonical"
```

Comment:
@oliviertassinari  I don't understand this. This means every older version of material-ui.com should redirect to the current version. However, we don't have a guarentee that either the content is equal nor that the links work. As far as I can tell this should only be used for technical reasons

> If you are unsure whether to do a 301 redirect or set a canonical, what should you do?

-- https://yoast.com/rel-canonical/

And I think this goes against providing multiple versions of the docs. Either we provide them  or not. But providing and then redirecting seems like not actually doing it? Same applies to the `alternates`. They should point to the current domain i.e. version, no?

Versioned docs comparison (there's space for more in the table, feel free to edit):
`document.querySelector('link[rel="canonical"]')`

| site                                                    | link rel canonical |
|---------------------------------------------------------|--------------------|
| https://react-legacy.netlify.com/                       | no                 |
| https://1.semantic-ui.com/                              | no                 |
| https://react-redux.js.org/5.x/introduction/quick-start | no                 |
| https://jaredpalmer.com/formik/docs/1.2.0/overview      | no                 |
| https://redux-form.com/7.3.0/docs/api/                  | no                 |
| https://1x.ant.design/                                  | no                 |
|                                                         |                    |
|                                                         |                    |
|                                                         |                    |
|                                                         |                    |
|                                                         |                    |
|                                                         |                    |
|                                                         |                    |
|                                                         |                    |

Reply:
Before covering the implementation, can we agree on the expected behavior? Should have all the older versions of the documentation indexed in Google and Algolia? (v3.1.0, v3.2.0, v3.3.0…).
For Algolia, it would be costly from a crawling perspective, it also seems wasteful to have them crawl 1-10k pages when 1k cover 80% of the need. For Google, we have a crawling budget, I would have the same underlying problem than with Algolia, does it worth spending some of our crawling budgets on almost never used pages? (to be confirmed with G.A. stats)
I would also fear the SEO "juice" dilution impact.

For these reasons, I was proposing to consolidate all the v3.x subdomain under a **single one**, exactly like we do with v0.x. For this consolidated subdomain, we can easily fix the warning link, fix the Algolia search, and, why not, enable Google indexation.

The nice impact of the current canonical logic is that it prevents all the Netlify previews to be indexed in Google.

Reply:
> Should have all the older versions of the documentation indexed in Google and Algolia? (v3.1.0, v3.2.0, v3.3.0, v3.4.0, …, v4.1.0, v4.2.0, etc.).

Of course. I think we agree on that. Otherwise the older versions become useless.

> For Algolia, it would be costly from a crawling perspective, it also seems wasteful to have them crawl 1-10k pages when 1k cover 80% of the need.

I would hope algolia only crawls on update. There's no reason to crawl periodically on pages that didn't change.

> I would also fear the SEO "juice" dilution impact.

Where is the evidence for that? It confirms everything I think about so called "SEO experts". They don't care about their users or the web in general. Just give me results even if they're useless. And then they need to be hired every few months when google penalizes their bad practices. No thank you. I rather a fewer hits that are useful than many hits that are useless.

> The nice impact of the current canonical logic is that it prevents all the Netlify previews to be indexed in Google.

I never saw a netlify preview of anything in google search results

Reply:
For the SEO benchmark, can we have a look at the top performers, the projects that have a good SEO ranking? Maybe Elasticsearch, Tensorflow, Rails, I don't know, I'm just guessing the projects that might be interested in copying.

Reply:
> Maybe Elasticsearch, Tensorflow, Rails, else?

Updated the table. Only tensorflow uses it but not to link to the latest version.

> the projects that have a great SEO ranking

What metrics do you use to determine that. I have now 10 sites that are similar to ours or are top performers in your eyes. None use rel="canonical" or if they do not like we do.